### PR TITLE
crypto-primitives no-std

### PR DESCRIPTION
**Pull request by debris**
major changes:

- `crypto-primitives` works with `no-std`
- moved `crypto-primitives/benches` to a new crate `cp-benches`, because of `criterion` dependency

### CODE REVIEW

File: crypto-primitives/src/merkle_tree/mod.rs [hunk 1]
```diff
@@ -255,23 +255,27 @@ impl std::fmt::Display for MerkleTreeError {
...
 fn log2(number: usize) -> usize {
-    (number as f64).log2() as usize
+    algebra::log2(number) as usize
 }
 
 /// Returns the height of the tree, given the size of the tree.
 #[inline]
 fn tree_height(tree_size: usize) -> usize {
-    log2(tree_size + 1)
+    log2(tree_size)
```
Comment (debris) [inline crypto-primitives/src/merkle_tree/mod.rs:278]: `algebra::log2` returns approximate value of `log2`. To make tests pass, I had to remove `+1` from here. I am not sure whether this changes the logic significantly

Comment (Pratyush) [reply crypto-primitives/src/merkle_tree/mod.rs:278]: Maybe we can use this: https://users.rust-lang.org/t/logarithm-of-integers/8506/5

Comment (Pratyush) [reply crypto-primitives/src/merkle_tree/mod.rs:278]: Nvm I should have read the code before commenting 🤦🏾‍♂️

File: crypto-primitives/src/crh/bowe_hopwood/mod.rs [hunk 2]
```diff
@@ -149,6 +152,32 @@ impl<G: Group, W: PedersenWindow> FixedLengthCRH for BoweHopwoodPedersenCRH<G, W
                     .reduce(G::zero, |a, b| a + &b)
             })
             .reduce(G::zero, |a, b| a + &b);
+
```
Comment (Pratyush) [inline crypto-primitives/src/crh/bowe_hopwood/mod.rs]: You can avoid duplication by doing `.sum()`; the latest master adds in support for this :)

File: crypto-primitives/src/crh/pedersen/mod.rs [hunk 3]
```diff
@@ -112,6 +114,22 @@ impl<G: Group, W: PedersenWindow> FixedLengthCRH for PedersenCRH<G, W> {
                 encoded
             })
             .reduce(G::zero, |a, b| a + &b);
+
+        #[cfg(not(feature = "parallel"))]
```
Comment (Pratyush) [inline crypto-primitives/src/crh/pedersen/mod.rs]: Same here

**Review by Pratyush — CHANGES_REQUESTED** [review]
Just the small changes to avoid duplication, and we should be good to go! Thanks for the PR!

**Comment by debris** [comment]
ready for another review :)

**Comment by Pratyush** [comment]
Looks good, thanks!

**Pull request merged**